### DISCUSSION TITLE
Dependency group vs extras

### DISCUSSION THREAD

**Discussion by albin02t** [python-poetry/poetry#6967] [Q&A]
I wanted to implement a new functionality as an extra using poetry and was going through the documentation on poetry. i had come across optional dependency groups and the extras concept in poetry and got confused between the two. What is the difference between adding an optional dependency group with all the dependencies for that feature or adding the deps as optional and creating an optional for it.

Also in the documentation this was mentioned which isn't clear. What does it mean to be removed in the below text:

> Any extras you don’t specify will be removed. Note this behavior is different from optional dependency groups not selected for install, e.g. those not specified via install --with.

**Comment by Young-Lord** [comment]
> Do I need it this dependencies during runtime or only during development? For the first case use [extras](https://python-poetry.org/docs/master/pyproject/#extras), for the later use [dependency groups](https://python-poetry.org/docs/master/managing-dependencies/#dependency-groups).

From official answer: https://github.com/python-poetry/poetry/issues/4891#issuecomment-993816696
